controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -713,6 +715,21 @@ private void handleStreamMetadataExists(final long timestamp, CompletableFuture<
         }
     }
 
+    @Override
+    CompletableFuture<Map<String, Data<Integer>>> getTxnInEpoch(int epoch) {
+        synchronized (txnsLock) {
+            Set<String> transactions = epochTxnMap.get(epoch);
+            Map<String, Data<Integer>> map;
+            if (transactions != null) {
+                map = activeTxns.entrySet().stream().filter(x -> transactions.contains(x.getKey()))
+                        .collect(Collectors.toMap(Map.Entry::getKey, x -> copy(x.getValue())));
+            } else {
+                map = Collections.emptyMap();
+            }
+            return CompletableFuture.completedFuture(Collections.unmodifiableMap(map));
```

Comment:
You only need to wrap the map in the `if` block; the `else` block already returns an immutable map.

Reply:
done